### DISCUSSION TITLE
How are Permissions and Polices linked?

### DISCUSSION THREAD

**Discussion by laggingreflex** [strapi/strapi#6295] [Legacy Topics]
I have a collection type "**Content**". By default the API route `GET /contents` gives a `403 Forbidden`. 

In order to make it public, I see there are two ways:

1. Set permissions in the Admin Panel > Roles & Permissions. 

2. Remove the `policies` field in `./api/content/config/routes.json`

    ```json
    {
      "routes": [
        {
          "method": "GET",
          "path": "/contents",
          "handler": "content.find",
          "config": {
           -"policies": []
          }
        },
      ]
    }
    ```

Since both these ways achieve the same result, I'm wondering how they're linked?

I'm also puzzled why having a seemingly empty `"policies": []` array doesn't automatically makes it public to begin with? It seems just having the `policies` key makes it ***depend*** on Permissions. Is that so?

**Comment by derrickmehaffy** [comment]
Effectively they aren't, the current "flow" at a high level would be:

1. User makes a request (with or without a JWT)
2. Request is sent through any middleware, then onto the users-permissions plugin to validate the user has access (validating the JWT, user/password, ect)
3. Request is then checked against the routes, if a policy exists it is ran. Based on where the code is, either before or after the `next()` it will be ran. Each `next()` means to move onto the next policy or at the end run the controller. Policies are useful for checking specific of the request (think field level permissions)
4. After all the polices have ran, and if no error is returned to the user, the controller will be ran.
5. Core controllers execute the services thus they are ran next
6. Services execute queries which make the database calls to the ORM
7. At this point the data is sanitized and so long as none of the policies had code after a `next()` then the data is returned to the user.

**Reply by derrickmehaffy** [reply]
(I do realize there are policy options on the permissions themselves but I have never seen that used, maybe @alexandrebodin or @lauriejim can elaborate on that)
